src/main.cpp

```diff
@@ -1299,8 +1299,7 @@ bool VerifyAmounts(const CCoinsViewCache& cache, const CTransaction& tx, std::ve
                 if (!MoneyRange(val.GetAmount()))
                     return false;
 
-                if (val.GetAmount() == 0)
-                    continue;
+                assert(val.GetAmount() != 0);
```

Comment:
An input should never happen like this if the logic is correct on the
output side. I thought an assert made the most sense here since it shouldn't be violated.